forge-gui-desktop/src/test/java/forge/ai/simulation/SpellAbilityPickerSimulationTest.java

```diff
@@ -371,6 +375,84 @@ public void targetRainbowLandOverDual() {
         AssertJUnit.assertEquals(desired, sa.getTargetCard());
     }
 
+    @Test
+    public void ensureAllLandsArePlayable() {
+        initAndCreateGame();
+
+        System.out.println("Adding lands to hand");
+
+        // add every land to the player's hand
+//        SimulationController.MAX_DEPTH = 0;
+        List<Card> funky = new ArrayList<>();
+        String previous = "";
+        for (PaperCard c : FModel.getMagicDb().getCommonCards().getAllCards()) {
+            // reset the game
+            Game game = resetGame();
+            Player p = game.getPlayers().get(1);
+            Player opponent = game.getPlayers().get(0);
+            opponent.setLife(20, null);
+
+            // add one of each basic to the battlefield so that bouncelands and similar work
+            addCard("Plains", p);
+            addCard("Island", p);
+            addCard("Swamp", p);
+            addCard("Mountain", p);
+            addCard("Forest", p);
+            // Add basics to library to ensure fetches work
+            addCardToZone("Plains", p, ZoneType.Library);
+            addCardToZone("Island", p, ZoneType.Library);
+            addCardToZone("Swamp", p, ZoneType.Library);
+            addCardToZone("Mountain", p, ZoneType.Library);
+            addCardToZone("Forest", p, ZoneType.Library);
+
+            game.getPhaseHandler().devModeSet(PhaseType.MAIN1, p);
+            game.getAction().checkStateEffects(true);
+
+            // Only add one version at a time
+            if (c.getName().equals(previous)) {
+                continue;
+            }
+            previous = c.getName();
+            if (c.getRules().getType().isLand()) {
+                // Skip glacial chasm, it's really weird.
+                if (c.getName().equals("Glacial Chasm")) {
```

Comment:
I will think about Glacial Chasm and other cards like the Lair Lands with:
> When Crosis’s Catacombs enters the battlefield, sacrifice it unless you return a non-Lair land you control to its owner’s hand.

or Lotus Vale and Lotus Field that has sacrifice costs